### PR TITLE
[RFC] Unify http utils for github and gitlab

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
experiment - unifies the http utils of gitlab/bitbucket, followup to #109

### CODE REVIEW

**Comment by asottile** [comment]
seems fine, it's an api break though (but an easy one to fix?) -- can you split them apart?

**Comment by RonnyPfannschmidt** [comment]
@asottile let me rebase onto upstream, then the dissonance should disappear

File: tests/gitlab_api_test.py [hunk 1]
```diff
@@ -1,22 +0,0 @@
-from all_repos.gitlab_api import get_all
```
Comment (asottile) [inline tests/gitlab_api_test.py:1]: this file is still there but empty?

Comment (RonnyPfannschmidt) [reply tests/gitlab_api_test.py:1]: good catch

**Comment by RonnyPfannschmidt** [comment]
closing this one, i think a different approach for unification is needed, will come up with a different experiment

**Pull request closed without merging**